plugin/grpctrace/interceptor_test.go

```diff
@@ -15,127 +15,291 @@ package grpctrace
+
+	checks := []struct {
+		name         string
+		expectedAttr map[core.Key]core.Value
+		eventsAttr   []map[core.Key]core.Value
+	}{
 		{
-			"FullyQualifiedMethodName",
-			"serviceName",
-			"/github.com.foo.%s/bar",
+			name: "/github.com.serviceName/bar",
+			expectedAttr: map[core.Key]core.Value{
+				rpcServiceKey:  core.String("serviceName"),
+				netPeerIPKey:   core.String("fake"),
+				netPeerPortKey: core.String("connection"),
+			},
+			eventsAttr: []map[core.Key]core.Value{
+				{
+					messageTypeKey: core.String("SENT"),
+					messageIDKey:   core.Int(1),
+				},
+				{
+					messageTypeKey: core.String("RECEIVED"),
+					messageIDKey:   core.Int(1),
+				},
+			},
 		},
 		{
-			"SimpleMethodName",
-			"serviceName",
-			"/%s/bar",
+			name: "/serviceName/bar",
+			expectedAttr: map[core.Key]core.Value{
+				rpcServiceKey: core.String("serviceName"),
+			},
+			eventsAttr: []map[core.Key]core.Value{
+				{
+					messageTypeKey: core.String("SENT"),
+					messageIDKey:   core.Int(1),
+				},
+				{
+					messageTypeKey: core.String("RECEIVED"),
+					messageIDKey:   core.Int(1),
+				},
+			},
 		},
 		{
-			"MethodNameWithoutFullPath",
-			"serviceName",
-			"%s/bar",
+			name:         "serviceName/bar",
+			expectedAttr: map[core.Key]core.Value{rpcServiceKey: core.String("serviceName")},
 		},
 		{
-			"InvalidMethodName",
-			"",
-			"invalidName",
+			name:         "invalidName",
+			expectedAttr: map[core.Key]core.Value{rpcServiceKey: core.String("")},
 		},
 		{
-			"NonAlphanumericMethodName",
-			"serviceName_123",
-			"/github.com.foo.%s/method",
+			name:         "/github.com.foo.serviceName_123/method",
+			expectedAttr: map[core.Key]core.Value{rpcServiceKey: core.String("serviceName_123")},
 		},
 	}
 
-	for _, tc := range testCases {
-		t.Run(tc.testName, tc.testUCISetsExpectedNameAttribute)
+	for _, check := range checks {
+		err = unaryInterceptor(context.Background(), check.name, req, reply, clientConn, uniInterceptorInvoker.invoker)
+		if err != nil {
+			t.Fatalf("failed to run unary interceptor: %v", err)
+		}
+
+		spanData, ok := exp.spanMap[check.name]
+		if !ok || len(spanData) == 0 {
+			t.Fatalf("no span data found for name < %s >", check.name)
+		}
+
+		attrs := spanData[0].Attributes
+		for _, attr := range attrs {
+			expectedAttr, ok := check.expectedAttr[attr.Key]
+			if ok {
+				if expectedAttr != attr.Value {
+					t.Errorf("name: %s invalid %s found. expected %s, actual %s", check.name, string(attr.Key),
+						expectedAttr.AsString(), attr.Value.AsString())
+				}
+				delete(check.expectedAttr, attr.Key)
+			}
+		}
+
+		// Check if any expected attr not seen
+		if len(check.expectedAttr) > 0 {
+			for attr := range check.expectedAttr {
+				t.Errorf("missing attribute %s in span", string(attr))
+			}
+		}
+
+		events := spanData[0].MessageEvents
+		for event := 0; event < len(check.eventsAttr); event++ {
+			for _, attr := range events[event].Attributes {
```

Comment:
If `len(check.eventsAttr) > len(events)` this will panic. Lengths should be determine equal before this.